kentikapi/internal/api_resources/saved_filters_test.go

```diff
@@ -1 +1,360 @@
+							"operator": "=",
+							"filterValue": "5"
+						}
+					],
+					"not": false
+				}
+			]
+		},
+		"filter_name": "test_filter",
+		"filter_description": "This is test filter description1",
+		"cdate": "2021-02-08T11:59:22.272Z",
+		"edate": "2021-03-04T08:46:12.322Z",
+		"filter_level": "company"
+	}`
+	expected := models.SavedFilter{
+		ID:        8275,
+		CompanyID: 74333,
+		Filters: models.Filters{
+			Connector: "All",
+			FilterGroups: []models.FilterGroups{
+				{
+					Connector: "Any",
+					Filters: []models.Filter{
+						{
+							FilterField: "dst_as",
+							Operator:    "=",
+							FilterValue: "5",
+						},
+					},
+					Not: false,
+				},
+			},
+		},
+		FilterName:        "test_filter",
+		FilterDescription: "This is test filter description1",
+		CreatedDate:       time.Date(2021, 2, 8, 11, 59, 22, 272e6, time.UTC),
+		UpdatedDate:       time.Date(2021, 3, 4, 8, 46, 12, 322e6, time.UTC),
+		FilterLevel:       "company",
+	}
+
+	transport := &api_connection.StubTransport{ResponseBody: getResponsePayload}
+	savedFiltersAPI := api_resources.NewSavedFiltersAPI(transport)
+
+	savedFilter, err := savedFiltersAPI.Get(context.Background(), 8275)
+
+	assert.NoError(t, err)
+	assert.Equal(t, &expected, savedFilter)
+	assert.Empty(t, transport.RequestBody)
+	assert.Equal(t, "/saved-filter/custom/8275", transport.RequestPath)
+}
+
+func TestCreateSavedFilter(t *testing.T) {
+	postResponsePayload := `
+	{
+        "filter_name":"test_filter1",
+        "filter_description":"This is test filter description",
+        "filters": {
+            "connector":"All",
+            "filterGroups": [
+                {
+                    "connector":"All",
+                    "filters": [
+                        {
+                            "filterField":"dst_as",
+                            "filterValue":"81",
+                            "operator":"="
+                        }],
+                    "not":false
+                }]
+            },
+        "company_id":"74333",
+        "filter_level":"company",
+        "edate":"2020-12-26T10:46:13.095Z",
+        "cdate":"2020-12-16T10:46:13.095Z",
+        "id":8152
+    }`
+	expected := models.SavedFilter{
+		FilterName:        "test_filter1",
+		FilterDescription: "This is test filter description",
+		Filters: models.Filters{
+			Connector: "All",
+			FilterGroups: []models.FilterGroups{
+				{
+					Connector: "All",
+					Filters: []models.Filter{
+						{
+							FilterField: "dst_as",
+							FilterValue: "81",
+							Operator:    "=",
+						},
+					},
+					Not: false,
+				},
+			},
+		},
+		CompanyID:   74333,
+		FilterLevel: "company",
+		CreatedDate: time.Date(2020, 12, 16, 10, 46, 13, 95e6, time.UTC),
+		UpdatedDate: time.Date(2020, 12, 26, 10, 46, 13, 95e6, time.UTC),
+		ID:          8152,
+	}
+	expectedRequestPayload := "{\"filter_name\":\"test_filter1\",\"filter_description\":\"This is test filter description\",\"cdate\":\"0001-01-01T00:00:00Z\"," +
```

Comment:
This is not readable for me. And it also tests the order of keys in the JSON, which is not wanted and too strict. I think the better approach is to use:
1. JSONPayloadInspector like in most of tests (we have a ticket to make it more strict, BTW: https://clkentik.atlassian.net/browse/KNTK-183)
2. unmarshal transport.RequestBody to interface{} and compare with expected map[string]interface{} (see: tags_integration_test.go -> TestClient_CreateTag)

The same in TestUpdateSavedFilter 

You could add TODO if you feel you're time-restricted.

Reply:
Ok I will add TODO